cli/src/cmd/forge/run.rs

```diff
@@ -96,11 +96,15 @@ impl Cmd for RunArgs {
         let bytecode = bytecode.expect("no bytecode for contract").object.into_bytes().unwrap();
         let needs_setup = abi.functions().any(|func| func.name == "setUp");
 
+        let env = evm_opts.evm_env();
+        // the db backend that serves all the data
+        let db = Backend::new(utils::get_fork(&evm_opts, &config.rpc_storage_caching), &env);
+
         let mut builder = ExecutorBuilder::new()
             .with_cheatcodes(evm_opts.ffi)
-            .with_config(evm_opts.evm_env())
-            .with_spec(crate::utils::evm_spec(&config.evm_version))
-            .with_fork(utils::get_fork(&evm_opts, &config.rpc_storage_caching));
+            .with_config(env)
```

Comment:
Oh also, unrelated, but I forgot to call `with_gas_limit` here which may be relevant if `forge run` forks: can you call `with_gas_limit(evm_opts.gas_limit())` as well?